test/fprime_gds/common/tools/seqgen_unit_test.py

```diff
@@ -1,7 +1,42 @@
+import filecmp
+from pathlib import Path
+import tempfile
 import unittest
 
+import fprime_gds.common.tools.seqgen as seqgen
 
 class APITestCases(unittest.TestCase):
     def test_nominal_sequence(self):
-
-        pass
+        self.assertEqual(self.diff_input_sequence(
+            "./input/simple_sequence.seq",
+            "./expected/simple_sequence_expected.bin",
+            "./resources/simple_dictionary.json"
+        ), True)
+    
+    # This test is expected to fail due to unmatched command in
+    # simple_bad_sequence.seq and seqgen should produce a SeqGenException. 
+    # Passing these executions is a test failure.
+    @unittest.expectedFailure 
+    def test_fail_unmatched_command_sequence(self):
+        self.diff_input_sequence(
+            "./input/simple_bad_sequence.seq",
+            "./expected/simple_sequence_expected.bin",
+            "./resources/simple_dictionary.json"
+        )
+        
+    def diff_input_sequence(self, input_sequence, expected_output, dictionary):
```

Comment:
This method name might be a little misleading, since you're proving an input sequence but diffing against the expected output sequence binary. Maybe consider rename